src/components/ShowList.jsx

```diff
@@ -31,11 +43,38 @@ function ShowList() {
       )}
       {value && value.docs.length > 0 && (
         <div>
-          Collection:
+          <input
+            type="text"
+            placeholder="Filter items..."
+            value={filter}
+            onChange={handleChange}
+          />
+          {filter && (
+            <button
+              onClick={() => setFilter('')}
+              aria-label="clear filter text"
+            >
+              <FontAwesomeIcon icon={['fas', 'times']} />
+            </button>
+          )}
           <ul>
-            {value.docs.map((doc) => (
-              <Item key={doc.id} {...doc.data()} id={doc.id} />
-            ))}
+            {filter
+              ? value.docs
+                  .map((doc) => ({ id: doc.id, ...doc.data() }))
```

Comment:
Nit: This is making us loop over the whole array twice since the filter is also looping over it. We _could_ remove this part and replace where we're accessing the fields with a call to `data()`.

```js
item.data().name.toLowerCase().includes(filter.toLowerCase()),
```

Reply:
I would second Kevin here. I don't understand why are we mapping through the items twice here. Can you please explain your thought process?

Reply:
We struggled with a way to render the items so that we could extract the id of the document - once you use .data() you can only see the data inside the document, and not the document id itself. 
At the moment, the Item component is using the doc id to make a call to the database so it needs to be passed as a prop when we render a shopping list item. Perhaps we can refactor this at a later date, as I think it will need some changes to Item also if we do? 🤔